src/terminal/init.lua

```diff
@@ -18,14 +18,38 @@ local M = {
   _DESCRIPTION = "Cross platform terminal library for Lua (Windows/Unix/Mac)",
 }
 
+--- Returns the height and width of the terminal screen.
+-- @treturn number, number the height in rows and the width in columns
+function M.get_dimensions()
+  local size = sys.get_terminal_size()
+  return size.height, size.width
+end
+
+--- Sets the cursor position and writes it to the terminal, without pushing onto the stack.
+-- @tparam number row
+-- @tparam number column
+-- @return true
+-- @within cursor_position
+function M.cursor_set(row, column)
+  -- Retrieve terminal dimensions
+  local height, width = M.get_dimensions()
+
+  -- Resolve negative indices using utils.resolve_index
+  row = utils.resolve_index(row, height)
+  column = utils.resolve_index(column, width)
+
+  -- Rest of the cursor_set logic
+  output.write("\27[" .. tostring(row) .. ";" .. tostring(column) .. "H")
+  return true
+end
+
 local pack, unpack do
   -- nil-safe versions of pack/unpack
   local oldunpack = _G.unpack or table.unpack -- luacheck: ignore
   pack = function(...) return { n = select("#", ...), ... } end
   unpack = function(t, i, j) return oldunpack(t, i or 1, j or t.n or #t) end
 end
 
-
```

Comment:
these whitespace changes are unrelated to this PR. I think it is better to not mix multiple issues in 1 PR. Can they be removed?